src/lib.rs

```diff
@@ -643,7 +643,9 @@ impl Node {
 	///
 	/// After this returns most API methods will return [`Error::NotRunning`].
 	pub fn stop(&self) -> Result<(), Error> {
-		let runtime = self.runtime.write().unwrap().take().ok_or(Error::NotRunning)?;
+		let mut runtime_lock = self.runtime.write().unwrap();
```

Comment:
Put commit msg explanation in code comment?

Reply:
Explaining why the runtime lock needs to be held until we are done shutting down isn't historical behavior?